src/main/java/com/glencoesoftware/ldaptool/Password.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (C) 2023 Glencoe Software, Inc. All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+package com.glencoesoftware.ldaptool;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.Callable;
+
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.Parameters;
+import picocli.CommandLine.ParentCommand;
+
+/**
+ * @author Chris Allan <callan@glencoesoftware.com>
+ */
+@Command(
+    name = "password",
+    mixinStandardHelpOptions = true,
+    exitCodeOnExecutionException = 100
+)
+public class Password implements Callable<Integer>
+{
+    private static final Logger log =
+            LoggerFactory.getLogger(Password.class);
+
+    @Option(
+        names = {"-p", "--password"},
+        description = "Passphrase",
+        interactive = true,
+        required = true
+    )
+    String password;
+
+    @Option(
+        names = {"--tries"},
+        description =
+            "Number of times to retry the password check " +
+            "(default: ${DEFAULT-VALUE})"
+    )
+    int tries = 1;
+
+    @Parameters(
+        index = "0",
+        description = "DN to check password for"
+    )
+    String dn;
+
+    @ParentCommand
+    private Main main;
+
+    Password() { }
+
+    @Override
+    public Integer call() throws Exception {
+        int exitCode = 0;
+        for (int i = 0; i < tries; i++) {
+            if (main.ldapImpl.validatePassword(dn, password)) {
+                System.out.println("Password check successful!");
+            } else {
+                System.out.println("Password check failed!");
+                exitCode = 1;
```

Comment:
For me the retry logic displayed unexpected results. Testing an existing user with incorrect password

```
[sbesson@sl2-topfx2-npsc ~]$  ./omero-ldaptool-0.2.1-SNAPSHOT/bin/omero-ldaptool password --config omero.config.20231003  sbesson  --tries 5 -p --log-level ERROR
Enter value for --password (Passphrase): 
Password check failed!
Password check successful!
Password check failed!
Password check successful!
Password check failed!
```

Reply:
Keep in mind it's the DN not the username you are passing. That's an artifact of the nature of the underlying API and not having to lookup users just to have things function. The fact it's ever successful I find quite amazing. Maybe try running with `--log-level=ALL` and see what it's actually doing?

Reply:
> Keep in mind it's the DN not the username you are passing.

Thanks, I had noted the requirement to use `DN` rather than `omeName` while reading the help but got distracted by the inconsistent error message.

So here's the current state of things using the latest commits as of [86886d2](https://github.com/glencoesoftware/omero-ldaptool/pull/6/commits/86886d2a972a9ac08307c4854a2c785be8780e4e).

- with a valid DN and an incorrect password

  ```
   [sbesson@sl2-topfx2-npsc ~]$ ./omero-ldaptool-0.2.1-SNAPSHOT/bin/omero-ldaptool password --config omero.config.20231003 "cn=sbesson,ou=edir,ou=people,ou=lifesci,o=dundee" --tries=5 
   2023-10-04 12:02:33,978 [main] WARN  ome.services.util.ReadOnlyStatus - not configured for read-only binary repository but assuming read-only anyway
   Enter passphrase for 'cn=sbesson,ou=edir,ou=people,ou=lifesci,o=dundee': 
   Password check failed!
   Password check failed!
   Password check failed!
   Password check failed!
   Password check failed!
   ```
   
- with an invalid DN 

  ```
   [sbesson@sl2-topfx2-npsc ~]$ ./omero-ldaptool-0.2.1-SNAPSHOT/bin/omero-ldaptool password --config omero.config.20231003 "cn=nonexisting,ou=edir,ou=people,ou=lifesci,o=dundee" --tries=5 
   2023-10-04 12:02:39,361 [main] WARN  ome.services.util.ReadOnlyStatus - not configured for read-only binary repository but assuming read-only anyway
   Enter passphrase for 'cn=nonexisting,ou=edir,ou=people,ou=lifesci,o=dundee': 
   Password check failed!
   Password check successful!
   Password check failed!
   Password check successful!
   Password check failed!
   ```
   
 Setting `--log-level ALL` does no provide much more infomrmation
 
```
Enter passphrase for 'cn=nonexisting,ou=edir,ou=people,ou=lifesci,o=dundee': 2023-10-04 12:04:52,010 [bitronix-async-jmx-worker] DEBUG b.tm.utils.ManagementRegistrar - Calling ManagementRegisterCommand on object with name bitronix.tm:type=JDBC,UniqueName=790fe401-3efc-4bd4-b24d-60918ba77962
2023-10-04 12:04:52,014 [bitronix-async-jmx-worker] DEBUG b.tm.utils.ManagementRegistrar - Calling ManagementRegisterCommand on object with name bitronix.tm:type=JDBC,UniqueName=790fe401-3efc-4bd4-b24d-60918ba77962,Id=0

2023-10-04 12:04:55,348 [main] DEBUG o.s.l.c.s.AbstractContextSource - Got Ldap context on server 'ldap://ldap.lifesci.dundee.ac.uk:389/ou=lifesci,o=dundee'
Password check failed!
2023-10-04 12:04:55,359 [main] DEBUG o.s.l.c.s.AbstractContextSource - Got Ldap context on server '<ldap_uri>'
Password check successful!
2023-10-04 12:04:55,361 [main] DEBUG o.s.l.c.s.AbstractContextSource - Got Ldap context on server '<ldap_uri>'
Password check failed!
2023-10-04 12:04:55,366 [main] DEBUG o.s.l.c.s.AbstractContextSource - Got Ldap context on server '<ldap_uri>'
Password check successful!
2023-10-04 12:04:55,368 [main] DEBUG o.s.l.c.s.AbstractContextSource - Got Ldap context on server '<ldap_uri>'
Password check failed!
```
   
Looking into enabling more low-level debugging to understand why we are getting these results